src/Approvals/SFA.DAS.Approvals/Application/Cohorts/Queries/GetSelectEmployer/GetSelectEmployerQueryHandler.cs

```diff
@@ -0,0 +1,198 @@
+
+        var accountProviderLegalEntities = providerRelationshipsResponse.AccountProviderLegalEntities
+            .Select(x => new AccountProviderLegalEntityItem
+            {
+                AccountId = x.AccountId,
+                AccountPublicHashedId = x.AccountPublicHashedId,
+                AccountHashedId = x.AccountHashedId,
+                AccountName = x.AccountName,
+                AccountLegalEntityId = x.AccountLegalEntityId,
+                AccountLegalEntityPublicHashedId = x.AccountLegalEntityPublicHashedId,
+                AccountLegalEntityName = x.AccountLegalEntityName,
+                AccountProviderId = x.AccountProviderId,
+                ApprenticeshipEmployerType = accountLevyStatusMap.GetValueOrDefault(x.AccountHashedId, "NonLevy")
+            })
+            .ToList();
+
+        if (!string.IsNullOrWhiteSpace(request.SearchTerm))
+        {
+            accountProviderLegalEntities = ApplySearch(accountProviderLegalEntities, request.SearchTerm);
+        }
+
+        if (!string.IsNullOrWhiteSpace(request.SortField))
+        {
+            accountProviderLegalEntities = ApplySort(accountProviderLegalEntities, request.SortField, request.ReverseSort);
+        }
+
+        var employers = accountProviderLegalEntities
+            .SelectMany(x => new[] { x.AccountLegalEntityName, x.AccountName })
+            .Where(x => !string.IsNullOrWhiteSpace(x))
+            .Distinct()
+            .ToList();
+
+        return new GetSelectEmployerQueryResult
+        {
+            AccountProviderLegalEntities = accountProviderLegalEntities,
+            Employers = employers
+        };
+    }
+
+    private async Task<Dictionary<string, string>> GetAccountLevyStatusMap(List<string> accountHashedIds)
+    {
+        var levyStatusMap = new Dictionary<string, string>();
+
+        if (!accountHashedIds.Any())
+        {
+            return levyStatusMap;
+        }
+
+        var accountTasks = accountHashedIds.Select(async accountHashedId =>
+        {
+            try
+            {
+                var accountResponse = await accountsApiClient.Get<GetAccountResponse>(
+                    new GetAccountRequest(accountHashedId));
+
+                return new
+                {
+                    AccountHashedId = accountHashedId,
+                    ApprenticeshipEmployerType = accountResponse?.ApprenticeshipEmployerType ?? "NonLevy"
+                };
+            }
+            catch (Exception ex)
+            {
+                logger.LogWarning(ex, "Failed to get account levy status for account {AccountHashedId}, defaulting to NonLevy", accountHashedId);
+                return new
+                {
+                    AccountHashedId = accountHashedId,
+                    ApprenticeshipEmployerType = "NonLevy"
+                };
+            }
+        });
+
+        var accountResults = await Task.WhenAll(accountTasks);
```

Comment:
I agree with the Parallel execution, but might it be wiser to use ForEachAsync() instead as this throttles the calls to the API. WhenAll() can consume all the working threads, but it will not be a problem if the list is small